cni-plugin/deployment/scripts/install-cni.sh

```diff
@@ -168,53 +192,79 @@ fi
+  #CNI_OLD_CONF_PATH="${CNI_OLD_CONF_PATH:-${CNI_CONF_PATH}}"
+  CNI_OLD_CONF_PATH="${CNI_CONF_PATH}"
+  CNI_CONF_PATH=$(find "${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}" -maxdepth 1 -type f \( -iname '*conflist' -o -iname '*conf' \) | grep -v "linkerd" | sort | head -n 1)
+  # If no files have been found, create our own.
+  CNI_CONF_PATH=${CNI_CONF_PATH:-"${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}/01-linkerd-cni.conf"}
+  CNI_CONF_FILE="${CNI_CONF_PATH}"
+  if [ -e "${CNI_CONF_FILE}" ]; then
+    # Add the linkerd-cni plugin to the existing list
+    CNI_TMP_CONF_DATA=$(cat "${TMP_CONF}")
+    CNI_CONF_DATA=$(jq --argjson CNI_TMP_CONF_DATA "$CNI_TMP_CONF_DATA" -f /linkerd/filter.jq "${CNI_CONF_FILE}")
+    echo "${CNI_CONF_DATA}" > ${TMP_CONF}
+  fi
 
-# Delete old CNI config files for upgrades.
-if [ "${CNI_CONF_PATH}" != "${CNI_OLD_CONF_PATH}" ]; then
-  echo "Removing CNI_OLD_CONF_PATH: ${CNI_OLD_CONF_PATH}"
-  rm -f "${CNI_OLD_CONF_PATH}"
-fi
+  # If the old config filename ends with .conf, rename it to .conflist, because it has changed to be a list
+  filename=${CNI_CONF_PATH##*/}
+  extension="${filename##*.}"
+  if [ "${filename}" != '01-linkerd-cni.conf' ] && [ "${extension}" = 'conf' ]; then
+    echo "Renaming ${CNI_CONF_PATH} extension to .conflist"
+    CNI_CONF_PATH="${CNI_CONF_PATH}list"
+  fi
+
+  # Delete old CNI config files for upgrades.
+  if [ "${CNI_CONF_PATH}" != "${CNI_OLD_CONF_PATH}" ]; then
+    echo "Removing CNI_OLD_CONF_PATH: ${CNI_OLD_CONF_PATH}"
+    rm -f "${CNI_OLD_CONF_PATH}"
+  fi
 
-# Move the temporary CNI config into place.
-mv "${TMP_CONF}" "${CNI_CONF_PATH}" || exit_with_error 'Failed to mv files.'
+  # Move the temporary CNI config into place.
+  mv "${TMP_CONF}" "${CNI_CONF_PATH}" || exit_with_error 'Failed to mv files.'
 
-echo "Created CNI config ${CNI_CONF_PATH}"
+  echo "Created CNI config ${CNI_CONF_PATH}"
+
+}
 
-# Unless told otherwise, sleep forever.
-# This prevents Kubernetes from restarting the pod repeatedly.
-should_sleep=${SLEEP:-"true"}
-echo "Done configuring CNI. Sleep=$should_sleep"
-while [ "${should_sleep}" = 'true'  ]; do
-  sleep infinity &
-  wait $!
-done
+install_cni_bin
+
+install_cni_conf
+
+# Start looping and watching fs events
+HOST_CNI_NET="${CONTAINER_MOUNT_PREFIX}${DEST_CNI_NET_DIR}"
+inotifywait -m ${HOST_CNI_NET} -e moved_to -e create -e delete |
+ while read -r directory action filename; do
+   if [[ "$filename" =~ .*.(conflist|conf) ]]; then 
+    filepath="$directory$filename"
+    if [ "$action" = "DELETE" ] && [ $filepath = "${CNI_CONF_PATH}" ]; then
+      echo "Detected change in $directory: $action $filename"
+      echo "Configuration file: ${CNI_CONF_PATH} removed; re-installing..."
+      install_cni_conf
+    elif [ $filepath != "${CNI_OLD_CONF_PATH}" ]; then
+      echo "Detected change in $directory: $action $filename"
+      echo "Detected new configuration file: $filepath; re-installing..."
+      install_cni_conf
+    fi
+  fi
+ done
```

Comment:
The indentation is a bit off here:

```suggestion
  while read -r directory action filename; do
    if [[ "$filename" =~ .*.(conflist|conf) ]]; then
      filepath="$directory$filename"
      if [ "$action" = "DELETE" ] && [ $filepath = "${CNI_CONF_PATH}" ]; then
        echo "Detected change in $directory: $action $filename"
        echo "Configuration file: ${CNI_CONF_PATH} removed; re-installing..."
        install_cni_conf
      elif [ $filepath != "${CNI_OLD_CONF_PATH}" ]; then
        echo "Detected change in $directory: $action $filename"
        echo "Detected new configuration file: $filepath; re-installing..."
        install_cni_conf
      fi
    fi
  done
```

Reply:
Fixed (I think). Didn't commit because I ended up re-writing that bit.